projects/packages/changelogger/src/Utils.php

```diff
@@ -149,33 +149,53 @@ public static function loadChangeFile( $filename, &$diagnostics = null ) {
 	}
 
 	/**
-	 * Get a timestamp for a file.
+	 * Get miscellaneous repo data for a file.
 	 *
-	 * @param string               $file File.
+	 * @param string               $file Filepath.
 	 * @param OutputInterface      $output OutputInterface to write debug output to.
 	 * @param DebugFormatterHelper $formatter Formatter to use to format debug output.
-	 * @return string|null
+	 * @param array|null           $input_options Options for the extraction.
+	 *   - add-pr-num: (bool) Whether to try to read a `(#12345)`-like PR number from the git commit creating each change entry. Default false.
+	 * @return array With keys 'timestamp' and 'pr-num'.
 	 */
-	public static function getTimestamp( $file, OutputInterface $output, DebugFormatterHelper $formatter ) {
+	public static function getRepoData( $file, OutputInterface $output, DebugFormatterHelper $formatter, $input_options = null ) {
+		$repo_data = array(
+			'timestamp' => null,
+			'pr-num'    => null,
+		);
+
 		try {
-			$process = self::runCommand( array( 'git', 'log', '-1', '--format=%cI', $file ), $output, $formatter );
+			$process = self::runCommand( array( 'git', 'log', '-1', '--format=%cI\n%s', $file ), $output, $formatter );
 			if ( $process->isSuccessful() ) {
-				$ret = trim( $process->getOutput() );
-				if ( preg_match( '/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:Z|[+-]\d{2}:\d{2})$/', $ret ) ) {
-					return $ret;
+				$cmd_output = explode( '\n', trim( $process->getOutput() ) );
```

Comment:
Then you'll also need to fix it here.
```suggestion
				$cmd_output = explode( "\n", trim( $process->getOutput() ) );
```